src/plan/global.rs

```diff
@@ -325,43 +273,13 @@ pub trait Plan: 'static + HasSpaces + Sync + Downcast {
         self.get_total_pages() - self.get_used_pages()
     }
 
-    fn is_emergency_collection(&self) -> bool {
-        self.base().emergency_collection.load(Ordering::Relaxed)
-    }
-
-    /// The application code has requested a collection. This is just a GC hint, and
-    /// we may ignore it.
-    ///
-    /// # Arguments
-    /// * `tls`: The mutator thread that requests the GC
-    /// * `force`: The request cannot be ignored (except for NoGC)
-    /// * `exhaustive`: The requested GC should be exhaustive. This is also a hint.
-    fn handle_user_collection_request(&self, tls: VMMutatorThread, force: bool, exhaustive: bool) {
-        // For exhaustive on generational plans, we force a full heap GC.
-        // A plan may implement this method themselves to handle the exhaustive GC.
-        if exhaustive {
-            if let Some(gen) = self.generational() {
-                gen.force_full_heap_collection();
-            }
-        }
-        self.base().handle_user_collection_request(tls, force)
-    }
-
     /// Return whether last GC was an exhaustive attempt to collect the heap.
     /// For example, for generational GCs, minor collection is not an exhaustive collection.
     /// For example, for Immix, fast collection (no defragmentation) is not an exhaustive collection.
     fn last_collection_was_exhaustive(&self) -> bool {
         true
     }
 
-    fn modify_check(&self, object: ObjectReference) {
```

Comment:
We expose this method to the binding, but the semantic of this method is very vague IMO. I think it is some legacy debugging code that can be traced back to the beginning of the project. Many bindings expose this function to native code, but I am not seeing any binding really uses it except JikesRVM. I suggest just removing it. If we do need such a method in the future, we can discuss and introduce a new method with clearer semantics.

I will need to update the bindings for the removing of this method.